src/main/java/com/epam/asset/tracking/service/BusinessProviderServiceImpl.java

```diff
@@ -18,6 +24,14 @@
 	@Autowired
 	BCryptPasswordEncoder bCryptPasswordEncoder;
 
+	@Autowired
+	BusinessProviderService businessProviderService;
+
+	@Autowired
+	JavaMailSender emailSender;
+
+	private String newPassword;
+

```

Comment:
The logic seems to be the same as in previous commits but in another class. I think the logic an be changed in order to avoid the problems I mention before with multiple threads